src/plugins/ucx/ucx_backend.cpp

```
@@ -137,16 +137,64 @@ class nixlUcxBackendReqH : public nixlBackendReqH {
         for (nixlUcxReq req : requests_) {
             const nixl_status_t ret = nixl::ucx::ucsToNixlStatus(ucp_request_check_status(req));
             if (ret == NIXL_IN_PROG) {
-                // TODO: Need process this properly.
-                // it may not be enough to cancel UCX request
                 worker_->reqCancel(req);
+                if (!drainRequest(req)) {
+                    // Still in flight past the deadline, so the request object is not
+                    // ours to reclaim: freeing it now lets a later completion write into
+                    // freed memory. Leak it rather than hand UCX a dangling request.
+                    NIXL_ERROR << "Request " << req
```

Cppcheck (2.21.0)

[style] The function 'getBase' is never used.

(unusedFunction)